cmd/minishift/cmd/delete.go

```diff
@@ -46,23 +48,41 @@ func runDelete(cmd *cobra.Command, args []string) {
 
 	util.ExitIfUndefined(api, constants.MachineName)
 
+	if err := cluster.UnRegister(api, minishiftConfig.InstanceConfig); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error unregistering the VM: %s", err.Error()))
+	}
+
+	if minishiftConfig.InstanceConfig.IsRegister {
```

Comment:
I think the problem is that this still does not compute in my mind w/o looking at the code for _cluster.UnRegister_. This can only work, because things are happening to minishiftConfig.InstanceConfig as you pass it into UnRegister. 

Why do you even pass _minishiftConfig.InstanceConfig_? Why not deal with the fact that we persist registration in instance config in the command itself, making this all explicit?